sdk/servicebus/azure-servicebus/azure/servicebus/management/_management_client.py

```diff
@@ -0,0 +1,287 @@
+
+    def create_queue(self, queue):
+        # type: (Union[str, QueueDescription]) -> QueueDescription
+        """Create a queue
+
+        :param queue: The queue name or a `QueueDescription` instance. When it's a str, it will be the name
+         of the created queue. Other properties of the created queue will have default values decided by the
+         ServiceBus. Use a `QueueDesceiption` if you want to set queue properties other than the queue name.
+        :type queue: Union[str, QueueDescription].
+        :returns: ~azure.servicebus.management.QueueDescription returned from ServiceBus.
+        """
+        try:
+            queue_name = queue.queue_name  # type: ignore
+            to_create = copy(queue)  # type: ignore
+            to_create.queue_name = None  # type: ignore
+        except AttributeError:
+            queue_name = queue  # type: ignore
+            to_create = QueueDescription()  # Use an empty queue description.
+
+        create_entity_body = CreateQueueBody(
+            content=CreateQueueBodyContent(
+                queue_description=to_create,  # type: ignore
+            )
+        )
+        request_body = create_entity_body.serialize(is_xml=True)
+        try:
+            with _handle_response_error():
+                et = cast(
+                    ElementTree,
+                    self._impl.queue.put(queue_name, request_body, api_version=constants.API_VERSION)  # type: ignore
+                )
+        except ValidationError:
+            # post-hoc try to give a somewhat-justifiable failure reason.
+            if isinstance(queue, (str, QueueDescription)):
+                raise_with_traceback(
+                    ValueError,
+                    message="queue must be a non-empty str or a QueueDescription with non-empty str queue_name")
+            raise_with_traceback(
+                TypeError,
+                message="queue must be a non-empty str or a QueueDescription with non-empty str queue_name")
+
+        return _convert_xml_to_object(queue_name, et, QueueDescription)  # type: ignore
+
+    def update_queue(self, queue_description):
+        # type: (QueueDescription) -> QueueDescription
+        """Update a queue
+
+        :param queue_description: The properties of this `QueueDescription` will be applied to the queue in
+         ServiceBus. Only a portion of properties can be updated.
+         Refer to https://docs.microsoft.com/en-us/rest/api/servicebus/update-queue.
+        :type queue_description: ~azure.servicebus.management.QueueDescription
+        :returns: ~azure.servicebus.management.QueueDescription returned from ServiceBus.
+        """
+
+        if not isinstance(queue_description, QueueDescription):
+            raise TypeError("queue_description must be of type QueueDescription")
+
+        to_update = QueueDescription()
+
+        for attr in QUEUE_DESCRIPTION_SERIALIZE_ATTRIBUTES:
+            setattr(to_update, attr, getattr(queue_description, attr, None))
```

Comment:
why do we need to create a new QueueDescrption and set each attribute instead of just using "queue_description" being passed in?

Reply:
I don't want to change the original passed in QueueDescription object

Reply:
gotcha, that makes sense to me.